torchtune/models/clip/_transforms.py

```diff
@@ -5,24 +5,131 @@
+        This will be used to generate possible_resolutions,
+        e.g. [(224, 224), (224, 448), (448, 224)] if max_num_tiles = 2 and tile_size = 224.
+        Default 4.
+    resample (str): Resampling method used when resizing images. Supports any enum of
+        ``torchvision.transforms.InterpolationMode``, e.g. "nearest", "nearest_exact", "bilinear", "bicubic".
+        Default 'bilinear'.
+    resize_to_max_canvas (bool): If true, the image will be upscaled without distortion to fit the largest possible
+        resolution from possible_resolutions.
+        If False, it will pick the resolution that minimizes downscaling, including no downscaling at all.
+        In this case, the image will only be upscaled if it's size < tile_size. Default False.
+    max_upscaling_size (int): Limit the upscaling performed on the image.
+    normalize (int): If true, image will be normalized using image_mean and image_std.
+    max_image_size (int): Max image size, used to bound the tensor sizes for export.
+    """
+
+    image_mean: Optional[List[float]] = None
+    image_std: Optional[List[float]] = None
+    possible_resolutions: Optional[List[Tuple[int, int]]] = None
+    tile_size: int = 224
+    max_num_tiles: Optional[int] = 4
+    resample: str = "bilinear"
+    resize_to_max_canvas: bool = False
+    max_upscaling_size: Optional[int] = None
+    normalize: bool = True
+    max_image_size: int = 5000
+
+
+class CLIPImageTransformCore(torch.nn.Module):
+    def __init__(self, config: ImageTransformConfig):
+        super().__init__()
+        self.config = config
+
+    """
+    Performs the core transformations involved in CLIPImageTransform;
+    1. Resize the image to target_size.
+    2. Pad the image to canvas_size.
+    3. Normalize the image using config.image_mean and config.image_std.
+    4. Reshape the image tensor into [n, channels, config.tile_size, config.tile_size].
+
+    Args:
+        image (torch.Tensor): image as a 3D tensor in form [C, H, W].
+        target_size (torch.Tensor): tensor of shape [1, 2] containing the target_height and target_width for resize.
+        canvas_size (torch.Tensor): tensor of shape [1, 2] containing the canvas_height and canvas_width for padding.
+
+    Returns:
+        Tuple[torch.Tensor, torch.Tensor]: Image tensor of shape [n, channels, config.tile_size, config.tile_size]
+            and aspect ratio tensor of shape [1, 2].
+    """
+
+    def forward(
+        self, image: torch.Tensor, target_size: torch.Tensor, canvas_size: torch.Tensor
+    ):
+        # Resize.
+        target_h, target_w = target_size.tolist()
+        torch._check(target_h > 1)
+        torch._check(target_h <= self.config.max_image_size)
+        torch._check(target_w > 1)
+        torch._check(target_w <= self.config.max_image_size)
+
+        image = resize(
+            image,
+            size=[target_h, target_w],
+            interpolation=self.config.resample,
+            antialias=True,
```

Comment:
Note, to lower to ExecuTorch we have to set this to False.

Reply:
I sent a message to someone in GenAI to see what they think